### ISSUE TITLE
Autowiring a GemfireTemplate into the application is not working in all cases.

### ISSUE THREAD

**Issue by kohlmu-pivotal** [spring-projects/spring-boot-data-geode#55]
Autowiring the {{o.s.d.g.GemfireTemplate}} as per the [documentation](https://docs.spring.io/spring-boot-data-geode-build/current/reference/html5/#geode-data-access-region-templates) does not work as documented.

Seems it has to do with the _lazy initialization_ of the Region.   If the _Repository_ is annotated wit `DependsOn("Customers")`, then the problem is resolved.

```
@Repository
public class DemoRepository {

	@Autowired
	@Qualifier("customersTemplate")
	private GemfireTemplate customersTemplate;

	public void putData(String key, String value) {
		customersTemplate.put(key, value);
	}
}
```

Fails with:

```
Description:

Field customersTemplate in com.example.demo.client.repo.DemoRepository required a bean of type 'org.springframework.data.gemfire.GemfireTemplate' that could not be found.


Action:

Consider defining a bean of type 'org.springframework.data.gemfire.GemfireTemplate' in your configuration.
```

**Comment by jxblum** [comment]
Need to merge to the `1.1.x` branch for the `1.1.2.RELEASE`.

**Comment by jxblum** [comment]
Merged to branch `1.1.x` for the `1.1.2.RELEASE`.
